### PR TITLE
ENH: BinaryMathematicalMorphology baseline testing

### PR DESCRIPTION
**Pull request by mseng10**
This enables baseline testing for the `BinaryMathematicalMorphology`
module. This specifically does the following:

- Defines `OutputBaseline` for all baseline tests in `Filtering/BinaryMathematicalMorphology/CMakeLists.txt`
- Implements WriteImage function for `ClosingBinaryImage`, `OpenBinaryImage` and `PruneBinaryImage`. This  that takes an Image and output filename as arguments and saves the output image to the output image filename.
- Saves new baseline images for `ClosingBinaryImage`, `OpenBinaryImage`, and
  `PruneBinaryIamge`.
- Changes original output `ClosingBinaryImage.png`, `OpenBinaryImage.png` and
  `PruneBinaryImage.png` to clarify these are outputs in the `QuickView`
window.
- Allow user to define input and output file paths for
  `ClosingBinaryImage`, `OpenBinaryImage`, `PruneBinaryImage`.

### CODE REVIEW

File: src/Filtering/BinaryMathematicalMorphology/ClosingBinaryImage/Code.cxx [hunk 1]
```diff
@@ -32,32 +33,35 @@ using ImageType = itk::Image<unsigned char, 2>;
 
 static void
 CreateImage(ImageType * const image);
+static void
+WriteImage(const ImageType::Pointer image, const std::string & fileName);
```
Comment (dzenanz) [inline src/Filtering/BinaryMathematicalMorphology/ClosingBinaryImage/Code.cxx:37]: This function was recently integrated into ITK. It is declared in `itkImageFileWriter.h`, see https://github.com/InsightSoftwareConsortium/ITK/commit/d8af2086421caabb8f7a47586e7acdc714ceb534.

File: src/Filtering/BinaryMathematicalMorphology/ClosingBinaryImage/Code.cxx [hunk 2]
```diff
@@ -91,6 +95,8 @@ main(int argc, char * argv[])
   viewer.AddImage(diff->GetOutput(), true, desc3.str());
   viewer.Visualize();
 #endif
+
+  WriteImage(closingFilter->GetOutput(), outputFilename);
```
Comment (dzenanz) [inline src/Filtering/BinaryMathematicalMorphology/ClosingBinaryImage/Code.cxx:99]:
```suggestion
  itk::WriteImage(closingFilter->GetOutput(), outputFilename);
```

**Review by dzenanz — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Comment by thewtex** [comment]
@mseng10 can this please be rebased on `master` for updated CI?

**Comment by mseng10** [comment]
@dzenanz @thewtex It seems the tests are failing on the Windows builds as the baseline images differ from the test's output images. Is there any reason why this would be working on the linux and macOS builds, but not Windows?

**Comment by dzenanz** [comment]
Looking at https://open.cdash.org/test/366062497, the differences are stark. It does not seem to work properly on Windows.

**Comment by mseng10** [comment]
> Looking at https://open.cdash.org/test/366062497, the differences are stark. It does not seem to work properly on Windows.

I see. I believe this may be a symptom of a bigger issue as I have started #212 and this is producing some similar results in the Windows builds (https://open.cdash.org/viewTest.php?onlyfailed&buildid=7117682). They work fine in Linux and MacOS. I will take a deeper dive into this to try and diagnose the issue.

**Comment by thewtex** [comment]
Rebasing on `master`.

**Comment by jhlegarreta** [comment]
b625917 rebased on `master` to solve conflicts.

**Comment by jhlegarreta** [comment]
Either the baselines seem to need an update:
https://open.cdash.org/viewTest.php?onlyfailed&buildid=7451836
e.g.
https://open.cdash.org/test/487486679

Or the test images do not look good.

**Comment by dzenanz** [comment]
This bug seems to affect many more examples. `image->Allocate();` has 149 hits while `image->FillBuffer(` has 64 hits.

**Comment by jhlegarreta** [comment]
Not sure why the doc build was cancelled. It seems like we cannot run just that one. The rest are passing.

**Comment by dzenanz** [comment]
[build-test-documentation (ubuntu-18.04)](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/actions/runs/1243892895):
System.TimeoutException: The HTTP request timed out after 00:01:40.

**Comment by dzenanz** [comment]
The input images are not the same. Pruning has an "erroneous" pixel.

You are right that closing should produce rounded corners.

**Comment by dzenanz** [comment]
I created #308 to keep track of the allocate issue.

**Pull request merged**